utils/src/spammer.rs

```diff
@@ -165,91 +187,120 @@ impl Spammer {
+        let txs_per_batch = self
+            .max_rate
+            .saturating_mul(self.batch_interval)
+            .checked_div(1000)
+            .unwrap_or(0);
+        debug!(
+            "Spamming {address} starting from nonce={latest_nonce} at rate {}, sending {txs_per_batch} txs every {}ms", 
+            self.max_rate,
+            self.batch_interval,
+        );
 
         // Initialize nonce and counters.
         let mut nonce = latest_nonce;
         let start_time = Instant::now();
         let mut txs_sent_total = 0u64;
-        let mut interval = time::interval(Duration::from_secs(1));
+        let mut interval = time::interval(Duration::from_millis(self.batch_interval));
+
         loop {
             // Wait for next one-second tick.
             let _ = interval.tick().await;
             let interval_start = Instant::now();
 
-            // Prepare batch of transactions for this interval.
-            let mut batch_entries = Vec::with_capacity(self.max_rate as usize);
+            // Verify the nonce for gaps
+            // TODO: probably this should run as a separate task
+            let on_chain_nonce = self.get_latest_nonce(address).await?;
+            // If the span between the on-chain nonce and the one we are about to send
+            // is too big, then probably there is a gap that doesn't allow the
+            // on-chain nonce too advance.
+            let nonce_span = nonce.saturating_sub(on_chain_nonce);
+            if nonce_span > self.max_rate {
+                debug!("Current nonce={nonce}, on-chain nonce={on_chain_nonce}. Sending 10 txs");
+                let batch_entries = self.build_batch_entries(10, on_chain_nonce).await?;
+                if let Some(results) = self.send_raw_batch(&batch_entries).await? {
+                    if results.len() != batch_entries.len() {
+                        return Err(eyre::eyre!(
+                            "Batch response count {} does not match request count {}",
+                            results.len(),
+                            batch_entries.len()
+                        ));
+                    }
 
-            for _ in 0..self.max_rate {
-                // Check exit conditions before creating each transaction.
-                if (self.max_num_txs > 0 && txs_sent_total >= self.max_num_txs)
-                    || (self.max_time > 0 && start_time.elapsed().as_secs() >= self.max_time)
-                {
-                    break;
+                    // Report individual results.
+                    for ((_, tx_bytes_len), result) in batch_entries.into_iter().zip(results) {
+                        let mapped_result = result.map(|_| tx_bytes_len);
+                        result_sender.send(mapped_result).await?;
+                    }
+                } else {
+                    debug!("Batch eth_sendRawTransaction timed out; skipping this tick");
+                    let _ = report_sender.send(interval_start).await;
```

Comment:
Why are we discarding the error?
And not something like:

```suggestion
                    report_sender.send(interval_start).await?;
```